components/cd-disclosure/cd-disclosure.js

```diff
@@ -0,0 +1,81 @@
+(function (Drupal) {
+  'use strict';
+
+  /**
+   * @see https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/
+   */
+  Drupal.behaviors.cdDisclosure = {
+    idPrefix: 'cd-disclosure--',
+
+    attach: function (context, settings) {
+      // Initialize toggable dropdown.
+      this.initializeDisclosures();
+    },
+
+    /**
+     * Set up all Disclosure widgets found in the DOM.
+     *
+     * For our purposes, the `disclosure` element is the wrapped content itself
+     * and the button to toggle it is generated automatically by this script.
+     */
+    initializeDisclosures: function () {
+      let disclosures = document.querySelectorAll('.cd-disclosure');
+
+      // All behavior is controlled by data attributes.
+      disclosures.forEach((disclosure) => {
+        // Set a random ID if none is found.
+        let disclosureId = disclosure.getAttribute('id') || this.idPrefix + Math.floor(Math.random(1) * 100000000);
+
+        // Apply ID if it wasn't there already.
+        if (!disclosure.getAttribute('id')) {
+          disclosure.setAttribute('id', disclosureId);
+        }
+
+        // Determine where to place the button.
+        let positionAfter = disclosure.dataset.cdDisclosureInsert === 'after' || false;
+
+        // Determine whether content is shown by default. We consider this the
+        // edge-case so omitting the attribute assumes it is hidden.
+        let contentVisible = disclosure.dataset.cdDisclosureShowByDefault === 'true' || false;
```

Comment:
We should set an initial `cd-disclosure--is-collapsed` or `cd-disclosure--is-expanded` on the element based on this value otherwise those classes are only set when pressing the button.

Also we probably want to remove the attribute.

See also comment about classes above.

Reply:
@rupl Bumping this comment.

To work well with the css rule in [components/cd-disclosure/cd-disclosure.css](https://github.com/UN-OCHA/common_design/pull/427/files#diff-f38e08f580263890eccc2b8e4683a8ac7107376e694ddd21f25a86b891bd1f1a), we should:

1. Remove the `data-cd-disclosure-show-by-default` if it is set to true
2. Set `cd-disclosure--is-collapsed` or `cd-disclosure--is-expanded` class depending on the value of the attribute

Reply:
Got it, sorry I missed this. Should be good now.